### PR TITLE
Perception: Added drops and their visualization

### PR DESCRIPTION
**Pull request by techoe**

### CODE REVIEW

File: modules/perception/camera/app/cipv_camera.cc [hunk 1]
```diff
@@ -864,21 +865,29 @@ bool Cipv::CollectDrops(const base::MotionBufferPtr &motion_buffer,
 
     object_id_skip_count_[cur_id] = 0;
 
+    auto pos = world2camera * obj->center;
+    float center_x = static_cast<float>(pos(2));
+    float center_y = static_cast<float>(-pos(0));
+
     object_trackjectories_[cur_id].push_back(
-        std::make_pair(obj->center(0), obj->center(1)));
+        std::make_pair(center_x, center_y));
```
Comment (xiaoxq) [inline modules/perception/camera/app/cipv_camera.cc]: `emplace_back(center_x, center_y);` is shorter.

File: modules/perception/camera/app/cipv_camera.cc [hunk 2]
```diff
@@ -864,21 +865,29 @@ bool Cipv::CollectDrops(const base::MotionBufferPtr &motion_buffer,
...
 
     if (debug_level_ >= 2) {
       AINFO << "object_trackjectories_[" << cur_id
             << " ].size(): " << object_trackjectories_[cur_id].size();
     }
 
-    //    obj->drops.clear();
+    Eigen::Matrix4f accum_motion_buffer =
+      motion_buffer->at(motion_size - 1).motion;
```
Comment (xiaoxq) [inline modules/perception/camera/app/cipv_camera.cc]: Two more indents.

File: modules/perception/camera/app/cipv_camera.cc [hunk 3]
```diff
@@ -887,9 +896,50 @@ bool Cipv::CollectDrops(const base::MotionBufferPtr &motion_buffer,
...
+              << accum_motion_buffer(1, 2) << ", "
+              << accum_motion_buffer(1, 3) << ", ";
+        AINFO << accum_motion_buffer(2, 0) << ", "
+              << accum_motion_buffer(2, 1) << ", "
+              << accum_motion_buffer(2, 2) << ", "
+              << accum_motion_buffer(2, 3) << ", ";
+        AINFO << accum_motion_buffer(3, 0) << ", "
+              << accum_motion_buffer(3, 1) << ", "
+              << accum_motion_buffer(3, 2) << ", "
+              << accum_motion_buffer(3, 3);
```
Comment (xiaoxq) [inline modules/perception/camera/app/cipv_camera.cc]: Better to add a helper function somewhere to pretty-print a Matrix4f, and then reuse it.

File: modules/perception/camera/tools/offline/visualizer.cc [hunk 4]
```diff
@@ -1058,44 +1096,99 @@ void Visualizer::Draw2Dand3D_all_info_single_camera(
           p_proj[i].y = static_cast<int>(proj[i](1) / proj[i](2));
         }
       }
+      if (object->b_cipv) {
+        cv::line(image_3D, p_proj[0], p_proj[1], color_cipv_,
+          cipv_line_thickness_);
```
Comment (xiaoxq) [inline modules/perception/camera/tools/offline/visualizer.cc]: Align with first parameter.

File: modules/perception/camera/tools/offline/visualizer.h [hunk 5]
```diff
@@ -59,12 +59,17 @@ class Visualizer {
...
-  void Draw2Dand3D_all_info_single_camera(const cv::Mat &img,
-                                          const CameraFrame &frame,
-                                          const Eigen::Matrix3d intrinsic,
-                                          const Eigen::Matrix4d extrinsic,
-                                          const Eigen::Affine3d &world2camera);
+  void Draw2Dand3D_all_info_single_camera(
+      const cv::Mat &img,
+      const CameraFrame &frame,
+      const Eigen::Matrix3d intrinsic,
+      const Eigen::Matrix4d extrinsic,
```
Comment (xiaoxq) [inline modules/perception/camera/tools/offline/visualizer.h]: Is it possible to be `const reference`?

File: modules/perception/camera/app/cipv_camera.cc (hunk 1 above)
Comment (techoe) [reply modules/perception/camera/app/cipv_camera.cc]: Thank you for the review and suggestion. Changed!

File: modules/perception/camera/app/cipv_camera.cc (hunk 2 above)
Comment (techoe) [reply modules/perception/camera/app/cipv_camera.cc]: Good to know. Thank you! Done.

File: modules/perception/camera/app/cipv_camera.cc (hunk 3 above)
Comment (techoe) [reply modules/perception/camera/app/cipv_camera.cc]: << supports print-out but it is ugly. Anyway I removed this and used original print-out.

File: modules/perception/camera/tools/offline/visualizer.cc (hunk 4 above)
Comment (techoe) [reply modules/perception/camera/tools/offline/visualizer.cc]: Indented all to be aligned with first parameter

File: modules/perception/camera/tools/offline/visualizer.h (hunk 5 above)
Comment (techoe) [reply modules/perception/camera/tools/offline/visualizer.h]: Yes. I changed them to "const reference"

**Pull request merged**